### PR TITLE
[DPS-XXXX] Require flags, fix generate order

### PR DESCRIPTION
**Pull request by gleb-lobov**

### CODE REVIEW

File: cmd/dp/generate.go [hunk 1]
```diff
@@ -45,6 +47,10 @@ Example:
 		dataproductDirectory, _ := cmd.Flags().GetString("data-products-directory")
 		dataProducts, _ := cmd.Flags().GetStringArray("data-product")
 
+		if len(sourceApps) == 0 && len(dataProducts) == 0 {
+			snplog.LogFatal(errors.New(("Please provide either --source-app or --source-app flag")))
+		}
+
```
Comment (gleb-lobov) [inline cmd/dp/generate.go:53]: Without it running `snowplow-cli dp generate` gives no output, as no loop below is run

File: cmd/dp/generate.go [hunk 2]
```diff
@@ -78,31 +84,28 @@ Example:
...
-			"eventSpecifications": []any{},
+func buildDpTpl(name string) model.CliResource[model.DataProductCanonicalData] {
+	return model.CliResource[model.DataProductCanonicalData]{
+		ApiVersion:   "v1",
+		ResourceType: "data-product",
+		ResourceName: uuid.NewString(),
+		Data: model.DataProductCanonicalData{
+			Name:                name,
+			SourceApplications:  []model.Ref{},
+			EventSpecifications: []model.EventSpecCanonical{},
```
Comment (gleb-lobov) [inline cmd/dp/generate.go:95]: Fixes the order 
```yaml
apiVersion: v1
data:
    appIds: []
    entities:
        enriched: []
        tracked: []
    name: my-sa
resourceName: 427dbfc3-cb25-41a8-81c8-3ad2145a2f17
resourceType: source-application
```

File: cmd/dp/generate.go (hunk 1 above)
Comment (jamessnowplow) [reply cmd/dp/generate.go:53]: i missed that. Should we check this with cobra in init()? `generateCmd.MarkFlagsOneRequired("source-app", "data-product")`

**Review by jamessnowplow — COMMENTED** [review]
i cant decide if cobra or by hand is best, maybe cobra?

File: cmd/dp/generate.go (hunk 1 above)
Comment (gleb-lobov) [reply cmd/dp/generate.go]: Done

**Review by jamessnowplow — APPROVED** [review]
👍

**Pull request merged**